### DISCUSSION TITLE
can i set the priority class to rook ceph operator ?

### DISCUSSION THREAD

**Discussion by RajuMedpur** [rook/rook#9789] [General]
can i set the operator priority class like below ? 

apiVersion: apps/v1
kind: Deployment
metadata:
  name: rook-ceph-operator
  namespace: {{ .Release.Namespace }}
  labels:
    operator: rook
    storage-backend: ceph
spec:
  selector:
    matchLabels:
      app: rook-ceph-operator
  replicas: 1
  template:
    metadata:
      labels:
        app: rook-ceph-operator
    spec:
       priorityClassName: {{ .Values.podPriority.priorityClassName }}    --> like this ?

**Comment by travisn** [comment]
Good timing on the question, I was just working on a fix for the issue in the helm chart #9620 to expose the priority class to the operator pod. The fix should be included in v1.8.6 later this week with #9790.

**Reply by RajuMedpur** [reply]
Thanks for the reply @travisn 

Currently which are all the pods supported for the pod priority in rook-ceph  ? (mon, osd, mgr , ??????)  ? any other pods , please let me know (i am using v1.6.2) 

i have some priority classes pre-defined and i have default low priority class which is set to all other pods for which pod priority is not present. 
in the above case . for the operator pod if i am not set the priority , it will take low priority and other pods will take high priority , 
is that fine ? (as of my knowledge operator pod is the one has  come first right ? )

**Reply by travisn** [reply]
All of the ceph daemons (mon, osd, mgr, mds, rgw) will have the priority classes set through the rook CRDs. Search the examples folder for that setting, or also see the [docs](https://rook.github.io/docs/rook/v1.8/ceph-cluster-crd.html#priority-class-names-configuration-settings). 

If the operator is a lower priority class that will be fine since it is not in the data path. After the cluster is configured, the operator is not really a critical component.
